modules/Microsoft.Network/networkManagers/connectivityConfigurations/deploy.bicep

```diff
@@ -0,0 +1,78 @@
+@sys.description('Conditional. The name of the parent network manager. Required if the template is used in a standalone deployment.')
+param networkManagerName string
+
+@maxLength(64)
+@sys.description('Required. The name of the connectivity configuration.')
+param name string
+
+@maxLength(500)
+@sys.description('Optional. A description of the connectivity configuration.')
+param description string = ''
+
+@sys.description('Required. Network Groups for the configuration.')
+param appliesToGroups array = []
+
+@allowed([
+  'HubAndSpoke'
+  'Mesh'
+])
+@sys.description('Required. Connectivity topology type.')
+param connectivityTopology string
+
+@sys.description('Conditional. List of hub items. This will create peerings between the specified hub and the virtual networks in the network group specified. Required if connectivityTopology is of type "HubAndSpoke".')
+param hubs array = []
+
+@allowed([
+  'True'
+  'False'
+])
+@sys.description('Optional. Flag if need to remove current existing peerings. If set to "True", all peerings on virtual networks in selected network groups will be removed and replaced with the peerings defined by this configuration. Optional when connectivityTopology is of type "HubAndSpoke". Default is "False".')
+param deleteExistingPeering string = 'False'
```

Comment:
Should we treat this as a boolean?

Reply:
Same for the other parameters

Reply:
Aligning to the template documentation and the API reference - It's a string. I would prefer to keep things as close as possible to avoid confusion.

Reply:
Hmm, this is indeed quite unfortunate. It just seems to unnecessary 'wrong' to me. I wonder why the RP did not implement a boolean when they had the chance. Also the name does not indicate that there'll ever be another value aside from 'true' & 'false'. If we however want to keep the string, I'd vouch to at least add an allowedSet, as silly as it seems.

@eriqua any thoughts from your end?

Reply:
> Hey @ahmadabdalla, we do in fact have similar 'odd' cases. For example the [AppGW](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.Network/applicationGateways/.test/common/deploy.test.bicep#:~:text=var%20appGWExpectedResourceID%20%3D%20%27%24%7BresourceGroup.id%7D/providers/Microsoft.Network/applicationGateways/%24%7BappGWName%7D%27) also has a child that wants to reference resource IDs of its linked resources. There, we introduced a variable `var appGWExpectedResourceID = '${resourceGroup.id}/providers/Microsoft.Network/applicationGateways/${appGWName}'` which we then use e.g. like `id: '${appGWExpectedResourceID}/frontendIPConfigurations/public'`. It's in any case quite - strange.

@AlexanderSehr will align to the approach above. At least we're in an agreement in principle :)

Reply:
We will have to do this conversion on every parameter in this resource type as unfortunately none of them are Booleans
![image](https://user-images.githubusercontent.com/28486158/210547345-dbba6928-3dcb-4017-84fc-74fdba992acb.png)

so would be something like this on each property:
```bicep
isGlobal: string(isGlobal)
```
I've already applied the 'allowed set' to each of those. @eriqua thoughts on this one please?

Reply:
@AlexanderSehr @eriqua Shall we resolve this comment? Its the only one left outstanding if there is no further feedback. Current implementation applies the allow set concept to restrict usage of the parameters.

Reply:
@eriqua is out of office this week, so let's give her a bit more time :)

Reply:
Sounds good to me. @ahmadabdalla? If you're fine with that we can resolve this comment and create the issue @eriqua proposed

Reply:
Happy with the suggestion. @eriqua @AlexanderSehr I've created an issue for it #2546.

Reply:
Great, thanks 😊